src/VaultSharp/V1/SecretsEngines/PKI/PKISecretsEngineProvider.cs

```diff
@@ -24,6 +24,17 @@ public async Task<Secret<CertificateCredentials>> GetCredentialsAsync(string pki
 
             return result;
         }
+        
+        public async Task<Secret<SignedCertificateData>> SignCertificateAsync(string pkiRoleName, SignCertificatesRequestOptions signCertificatesRequestOptions, string pkiBackendMountPoint = null, string wrapTimeToLive = null)
+        {
+            Checker.NotNull(pkiRoleName, "pkiRoleName");
+            Checker.NotNull(signCertificatesRequestOptions, "signCertificatesRequestOptions");
+
+            var result = await _polymath.MakeVaultApiRequest<Secret<SignedCertificateData>>(pkiBackendMountPoint ?? _polymath.VaultClientSettings.SecretsEngineMountPoints.PKI, "/sign/" + pkiRoleName, HttpMethod.Post, signCertificatesRequestOptions, wrapTimeToLive: wrapTimeToLive).ConfigureAwait(_polymath.VaultClientSettings.ContinueAsyncTasksOnCapturedContext);
+            result.Data.CertificateFormat = signCertificatesRequestOptions.CertificateFormat;
```

Comment:
This property is not marked with JsonIgnore... Either we should let json set it, or if we are explicitly setting it, it should be marked with json-ignore.

Reply:
CertificateFormat property is marked as JsonIgnore in AbstractCertificateData class. Or maybe I am referring to wrong property?

Reply:
got it. i was looking at the request poco by mistake.